app/src/NativeBase/Components/Project/ProjectRegisterInterest.tsx

```diff
@@ -0,0 +1,224 @@
+
+  const [responseHeader, setResponseHeader] = useState('')
+  const [responseMessage, setResponseMessage] = useState('')
+  const [modalVisible, setModalVisible] = useState(false)
+  const navigation = useNavigation()
+
+  useEffect(() => {
+    if (responseData || responseError) {
+      setLoading(false)
+      setModalVisible(true)
+
+      if (responseData) {
+        setSuccess(true)
+        setResponseHeader('Application Received')
+        setResponseMessage(
+          'Your request has been received. The STA team will respond shortly.',
+        )
+        navigation.setOptions({
+          title: 'Registration confirmed',
+        })
+      }
+      if (responseError) {
+        setResponseHeader('Something went wrong')
+        setResponseMessage(
+          "Sorry, we couldn't send your message - please try again. If this keeps happening, please contact the STA Volunteer App team.",
+        )
+      }
+    }
+  }, [responseData, responseError, navigation])
+
+  const onClose = () => {
+    setModalVisible(false)
+    goBack()
+  }
+  const validateField = (fieldName: string, value: string): boolean => {
+    let valid = true
+    let errorType = 'missing' as 'invalid' | 'missing'
+
+    switch (fieldName) {
+      case 'email':
+        if (!value) {
+          valid = false
+        } else if (!validateEmail(value)) {
+          valid = false
+          errorType = 'invalid'
+        }
+        break
+      default:
+        if (!value) {
+          valid = false
+        }
+        break
+    }
+
+    if (valid) {
+      setErrors(latestErrors => {
+        const updatedErrors = { ...latestErrors }
+        delete updatedErrors[fieldName]
+
+        return updatedErrors
+      })
+    } else {
+      setErrors(latestErrors => ({
+        ...latestErrors,
+        [fieldName]: {
+          type: errorType,
+        },
+      }))
+    }
+
+    return valid
+  }
+
+  const validateForm = (): boolean => {
+    const firstNameValid = validateField('firstName', firstName)
+    const lastNameValid = validateField('lastName', lastName)
+    const emailValid = validateField('email', email)
+
+    return firstNameValid && lastNameValid && emailValid
+  }
+
+  const submitForm = (): void => {
+    const formIsValid = validateForm()
+
+    if (formIsValid) {
+      setLoading(true)
+
+      registerInterest({
+        project: {
+          it_key: project.it_key,
+          res_id: project.res_id,
+        },
+        user: {
+          firstName,
+          lastName,
+          email,
+          lookingForPeerSupport,
+          availableFrom: dayjs(availableFromDate).format('YYYY-MM-DD'),
+        },
+      })
+    }
+  }
+  const { height } = Dimensions.get('window')
+
+  return (
+    <>
+      <ResponseModal
+        isOpen={modalVisible}
+        header={responseHeader}
+        message={responseMessage}
+        success={success}
+        onClose={onClose}
+      />
+
+      <VStack minHeight={height - 130} marginBottom="10">
```

Comment:
What is the `130` value doing here? (is it something to do with the height of the date picker? will it always be 130 on all screens?)  

Please could you set it earlier using a variable name e.g. `const someDescriptiveName = 130` to help someone else understand what it represents.

Reply:
It should be the height of topofapp + button, setting minheight in order to fill the empty space between the page contents and bottom of screen.  I don't think it should change?  It probably needs tests on ios though?